### PR TITLE
Add support for plain text only emails

### PR DESCRIPTION
**Pull request by leo-lb**
##### Summary
Fixes https://github.com/netdata/netdata/issues/6458
##### Component Name
Email alarms notifications
##### Additional Information
To enable the functionality add `EMAIL_PLAINTEXT_ONLY="YES"` in `health_alarm_notify.conf`

### CODE REVIEW

**Comment by leo-lb** [comment]
@cakrit The CLA is quite offensive, I don't want to allow my code to be re-used in proprietary software. This is a quite small change that in my opinion is not worth any money, but I certainly wont contribute anything more significant if the CLA does not guarantee me that the code will stay under FOSS licenses.

**Comment by cakrit** [comment]
> @cakrit The CLA is quite offensive, I don't want to allow my code to be re-used in proprietary software. This is a quite small change that in my opinion is not worth anything, but I certainly wont contribute anything more significant if the CLA does not guarantee me that the code will stay under FOSS licenses. Preferably it should be licensed under the AGPL.

I see how it could be misunderstood. The netdata agent is and forever will be FOSS. Discussing with @ktsaou the wording we have there, we definitely do not intend to ever make it proprietary!!!

File: health/notifications/alarm-notify.sh.in [hunk 1]
```diff
@@ -2093,6 +2094,8 @@ SENT_SYSLOG=$?
 # -----------------------------------------------------------------------------
 # send the email
 
+if [[ "${EMAIL_PLAINTEXT_ONLY}" == "YES" ]]; then
```
Comment (cakrit) [inline health/notifications/alarm-notify.sh.in]: You need `!=` here.

**Comment by thiagoftsm** [comment]
Hi @leo-lb ,

For you understand better what @cakrit is saying, let me show the results that I am receiving from your PR. Without to set any new variable in my health_alarm_notify.conf, I received the following e-mail body:

"--multipart-boundary
Content-Type: text/plain; encoding=ANSI_X3.4-1968
Content-Disposition: inline
Content-Transfer-Encoding: 8bit

ceres is critical

ram in use = 51.8% system RAM used"

On the other hand, when I set the variable that you created "EMAIL_PLAINTEXT_ONLY" I received the following e-mail body:

"Evaluated Expression :  $this > 5
Expression Variables :  [ $this = 52.7362019 ] 

The host has 0 WARNING and 1 CRITICAL alarm(s) raised.

--multipart-boundary
Content-Type: text/html; encoding=ANSI_X3.4-1968
Content-Disposition: inline
Content-Transfer-Encoding: 8bit

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; box-sizing: border-box; font-size: 14px; margin: 0; padding: 0;">
<body style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; font-size: 14px; width: 100% !important; min-height: 100%; line-height: 1.6; background: #f6f6f6; margin:0; padding: 0;">
<table>
    <tbody>
    <tr>
        <td style="vertical"

I am cutting this last e-mail, because it is really big. So as you can see your comparison is delivering the invert result.

Best regards!

**Comment by cakrit** [comment]
Regarding the CLA, we'll add a sentence in the note, stating that we intend to have the netdata agent be forever FOSS. 

Another necessary clarification is that the contribution is being made under GPLv3. So even if we ever changed to a different license, anyone could fork the entire repo on the day of the license change and continue on that fork. 

The main reason for the wording is that copyright transfer can't be conditional. A contributor either transfers the right or doesn't.

**Comment by leo-lb** [comment]
Sorry for the mistakes. They're all corrected now.

@cakrit I understand, and do hope that keeping this project forever FOSS is the intention. As you have made it now, the CLA is conditionnal I believe? As in, you can't change to a non-FOSS license without breaking the CLA. Thank you for trying to adapt legalese towards that goal.

**Comment by thiagoftsm** [comment]
Hi @leo-lb ,

You do not need to worry with the fact you do mistakes, because everybody does this, you would need to worry case you ignore them instead to fix, and this is not the case, because I tested your PR now and it worked as we were expecting ! Congratulations! 
Before I approve your PR I will only ask more one thing, as you can see in your script, you are defining the variables 'email_plaintext_part' and 'email_html_part' independent of the user choice, so my last suggestion for this PR is you define the variables only case they are really necessary. I think this will be my last request, because I am according with the rest of your implementation.

**Comment by leo-lb** [comment]
@thiagoftsm The PR was very much half assed and that is why I apologized. These variables were suggested by @Ferroin and they do increase readability because else I would have to write the same content twice, creating code bloat. So I will leave it like that. If you want to make text and html parts configurable, this would be a whole new feature in itself because there's a templating format with macros to create, to embed variables.

**Comment by thiagoftsm** [comment]
> @thiagoftsm The PR was very much half assed and that is why I apologized. These variables were suggested by @Ferroin and they do increase readability because else I would have to write the same content twice, creating code bloat. So I will leave it like that. If you want to make text and html parts configurable, this would be a whole new feature in itself because there's a templating format with macros to create, to embed variables.

All right, in my opinion they could be declared inside the respective conditional that it will be used, but considering that the performance gain will be very small, I will approve it, because the main goal was achieved in my tests. More one time, congratulation by your work!

**Comment by leo-lb** [comment]
@thiagoftsm I don't think you should worry about variable initialization performance when using Bash. Glad to help. When can this be merged?

**Comment by thiagoftsm** [comment]
Hi @leo-lb ,

I always try to save computer processing time, this is the motive I asked you that changes, but as I said it is not a real problem.
Today is Sunday yet for the majority of the team, so I think other coworkers will check your script during the week and we will merge it!

**Comment by leo-lb** [comment]
@thiagoftsm Moved the variable initialization around.

**Comment by cakrit** [comment]
> Sorry for the mistakes. They're all corrected now.
> 
> @cakrit I understand, and do hope that keeping this project forever FOSS is the intention. As you have made it now, the CLA is conditionnal I believe? As in, you can't change to a non-FOSS license without breaking the CLA. Thank you for trying to adapt legalese towards that goal.

We can't make copyright transfer conditional, as mentioned in the last sentence of https://github.com/netdata/netdata/pull/6485#issuecomment-513170602, but we can't envision any scenarios that would make us move away from FOSS for the agent. It just makes no sense, given the strategy explained in [our netdata cloud announcement](https://blog.netdata.cloud/posts/netdata-cloud-announcement/). 

We want the netdata agent to be installed everywhere and the only way to guarantee that is via FOSS.

**Comment by leo-lb** [comment]
> Hi @leo-lb ,
> 
> I tested your last changes and they are working fine again! :)
> I saw that you moved the HTML inside the "else" and this removed the HTML variable when it is not been used, something that I consider very good, because the computer works with a lot of bytes to mount that message.
> You did not move the plain text inside the if, but this is minor as we already talked.

The plain text version is *always* required, it will have to be initialized in ALL cases.

**Comment by thiagoftsm** [comment]
All right, we already merged it ! 
Thanks !

**Pull request merged**